pallets/sgx-account-linker/src/lib.rs

```diff
@@ -137,9 +139,20 @@ pub mod pallet {
 
 			let _ = ensure_signed(origin)?;
 
-			let current_block_number = <frame_system::Pallet<T>>::block_number();
-			ensure!(expiring_block_number > current_block_number, Error::<T>::LinkRequestExpired);
-			ensure!((expiring_block_number - current_block_number) < T::BlockNumber::from(EXPIRING_BLOCK_NUMBER_MAX),
+			// get the layer one block number from storage
+			let key = Self::storage_value_key("System", "LayerOneNumber");
```

Comment:
Could you add a comment here that this is only a temporary solution, or create a separate issue (or both)? Because having a hardcoded storage key value which needs to be the same as in the (independent) enclave-runtime is not nice. Maybe also include a link to the issue https://github.com/integritee-network/sgx-runtime/issues/27

Reply:
will add a comment and create a new issue.